File: backend/ModuMessenger/src/main/java/com/example/modumessenger/fcm/entity/FcmMessage.java
package com.example.modumessenger.fcm.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
@AllArgsConstructor
public class FcmMessage {
    private boolean validate_only;
    private Message message;

    @Builder
    @AllArgsConstructor
    @Data
    public static class Message {
        private Notification notification;
        private String token;
    }

    @Builder
    @AllArgsConstructor
    @Data
    public static class Notification {
        private String title;
        private String body;
        private String image;
    }
}

File: backend/push-service/src/main/kotlin/com/example/pushservice/fcm/entity/FcmMessage.kt
package com.example.pushservice.fcm.entity

/** FCM HTTP v1 요청 본문 모양(현재는 firebase-admin 을 써서 직접 쓰지 않는다). */
data class FcmMessage(
    @Suppress("PropertyName") val validate_only: Boolean = false,
    val message: Message? = null,
) {
    data class Message(val notification: Notification? = null, val token: String? = null)

    data class Notification(val title: String? = null, val body: String? = null, val image: String? = null)
}
